File: samples/snippets/visualbasic/VS_Snippets_CLR_System/system.math.round.overload/vb/mean1.vb
' Visual Basic .NET Document
Option Strict On

Module Mean
    Public Sub Example()
        ' <Snippet2>
        Dim values() As Decimal = {1.15D, 1.25D, 1.35D, 1.45D, 1.55D, 1.65D}
        Dim sum As Decimal

        ' Calculate true mean.
        For Each value In values
            sum += value
        Next
        Console.WriteLine("True mean:     {0:N2}", sum / values.Length)

        ' Calculate mean with rounding away from zero.
        sum = 0
        For Each value In values
            sum += Math.Round(value, 1, MidpointRounding.AwayFromZero)
        Next
        Console.WriteLine("AwayFromZero:  {0:N2}", sum / values.Length)

        ' Calculate mean with rounding to nearest.
        sum = 0
        For Each value In values
            sum += Math.Round(value, 1, MidpointRounding.ToEven)
        Next
        Console.WriteLine("ToEven:        {0:N2}", sum / values.Length)

        ' The example displays the following output:
        '       True mean:     1.40
        '       AwayFromZero:  1.45
        '       ToEven:        1.40
        ' </Snippet2>
    End Sub
End Module

File: snippets/fsharp/System/Decimal/Round/mean1.fs
module mean1

// <Snippet2>
open System

let values = [| 1.15m; 1.25m; 1.35m; 1.45m; 1.55m; 1.65m |]
let mutable sum = 0m

// Calculate true mean.
for value in values do
    sum <- sum + value

printfn $"True mean:     {sum / decimal values.Length:N2}"

// Calculate mean with rounding away from zero.
sum <- 0m
for value in values do
    sum <- sum + Math.Round(value, 1, MidpointRounding.AwayFromZero)

printfn $"AwayFromZero:  {sum / decimal values.Length:N2}"

// Calculate mean with rounding to nearest.
sum <- 0m
for value in values do
    sum <- sum + Math.Round(value, 1, MidpointRounding.ToEven)

printfn $"ToEven:        {sum / decimal values.Length:N2}"

// The example displays the following output:
//       True mean:     1.40
//       AwayFromZero:  1.45
//       ToEven:        1.40
// </Snippet2>